lib/generators/declare_schema/migration/migrator.rb

```diff
@@ -451,31 +451,68 @@ def change_indexes(model, old_table_name, to_rename)
               end
           end
 
-          drop_indexes = (existing_indexes_without_primary_key - model_indexes_without_primary_key).map do |i|
+          indexes_to_drop = existing_indexes_without_primary_key - model_indexes_without_primary_key
+          indexes_to_add = model_indexes_without_primary_key - existing_indexes_without_primary_key
+
+          renamed_indexes_to_drop = []
+          renamed_indexes_to_add = []
+
+          indexes_to_drop.each { |index_to_drop|
+            renamed_columns = index_to_drop.columns.map do |column|
+              to_rename[column]
```

Comment:
I'm assuming this hash lookup can sometimes return `nil`. Is that right? Will that cause a problem to have `nil` values in the array? If so, you could use `map_compact`.

Reply:
Sounds good. You could even use `.fetch` which takes a default value to use when the key is not found:
```
to_rename.fetch(column, column)
```

Reply:
I ended up using `.fetch`. Thanks!